credential/src/iam_task.rs

```diff
@@ -0,0 +1,111 @@
+use std::env;
+use std::io::Read;
+use std::time::Duration as StdDuration;
+
+use hyper::Client;
+use hyper::header::Connection;
+use serde_json::{self, Value};
+
+use {AwsCredentials, CredentialsError, ProvideAwsCredentials};
+
+const AWS_CREDENTIALS_PROVIDER_IP: &'static str = "169.254.170.2";
```

Comment:
Yeah I'm probably going to make changes to the other providers to bring them more in line with this one (refactoring, etc.).